File: Staircase.java


public class Staircase {
	
	public static void main(String[] args) {
		buildStairs(6);
	}
	
	public static void buildStairs(int stairs) {
		String space = " ";
		String sign = "#";
		
		StringBuilder stair = new StringBuilder();
		
		int flag = stairs;
		
		for(int i = 1; i <= stairs; i++) {
			flag = stairs - i;
			
			while(flag != 0) {
				stair.append(space);
				flag--;
			}
			
			while(stair.length() != stairs) {
				stair.append(sign);
			}
		
			print(stair.toString());
			stair.setLength(0);
		}	
	}
	
	public static void print(String stair) {
		System.out.println(stair);
	}
	
	
}

File: kotlin/Staircase.kt
/**
 *
 * @author slava on 7/5/19
 */
class Staircase {
    fun buildStairs(stairs: Int) {
        val space: String = " "
        val sign: String = "#"

        var stair: String = ""
        var flag: Int = stairs

        for(i in 1..stairs) {
            flag = stairs - i

            while (flag != 0) {
                stair += space
                flag--
            }

            while (stair.length != stairs) {
                stair += sign
            }
            println(stair)
            stair = ""
        }
    }
}

